scripts/tools/IndexRecreateRepopulate.php

```diff
@@ -0,0 +1,173 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\scripts\tools;
+
+use Elastic\Elasticsearch\Client;
+use oat\generis\model\DependencyInjection\ServiceOptions;
+use oat\generis\model\DependencyInjection\ServiceOptionsInterface;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\tao\scripts\tools\MigrationAction;
+use oat\taoAdvancedSearch\model\Resource\Task\ResourceMigrationTask;
+use oat\taoAdvancedSearch\model\SearchEngine\Contract\IndexerInterface;
+use oat\taoAdvancedSearch\model\SearchEngine\Driver\Elasticsearch\ElasticSearchClientFactory;
+use oat\taoAdvancedSearch\model\SearchEngine\Driver\Elasticsearch\ElasticSearchConfig;
+use oat\taoAdvancedSearch\model\SearchEngine\Service\IndexPrefixer;
+use RuntimeException;
+
+/**
+ * php index.php 'oat\taoAdvancedSearch\scripts\tools\IndexRecreateRepopulate' -i tests
+ */
+class IndexRecreateRepopulate extends ScriptAction
+{
+    private Client $client;
+    private IndexPrefixer $prefixer;
+
+    protected function provideUsage(): array
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Recreates and repopulates a particular index'
+        ];
+    }
+
+    protected function provideOptions(): array
+    {
+        return [
+            'index' => [
+                'prefix' => 'i',
+                'flag' => false,
+                'description' => 'The index to recreate',
+                'required' => true
+            ],
+            'limit' => [
+                'prefix' => 'l',
+                'longPrefix' => 'limit',
+                'flag' => false,
+                'description' => 'The limit of resources to be processed per index',
+                'defaultValue' => 50
+            ],
+        ];
+    }
+
+    protected function provideDescription(): string
+    {
+        return 'Recreate and repopulate a particular index';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function run(): Report
+    {
+        $report = Report::createInfo('Recreating and repopulating index...');
+
+        try {
+            /** @var ServiceOptionsInterface $serviceOptions */
+            $serviceOptions = $this->getServiceManager()->get(ServiceOptions::SERVICE_ID);
+            $this->prefixer = new IndexPrefixer(new ElasticSearchConfig($serviceOptions));
+            $this->client = (new ElasticSearchClientFactory(new ElasticSearchConfig($serviceOptions)))->create();
+
+            $index = $this->getOption('index');
+            $limit = $this->getOption('limit');
+            $classUri = $this->findClassUri($index);
+
+            $this->deleteIndex($index);
+            $this->createIndex($index);
```

Comment:
This will take some time for bigger data collection. Where do we want to execute this command?

Reply:
create index only create index itself without data
population will be done in queue

Reply:
Right, population can take some time, but will be done asynchronously. In premium UAT there is ~1000 tests and indexing finished under 5 minutes